src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -80,6 +76,9 @@ public GATKReadFilterPluginDescriptor(final List<ReadFilter> toolDefaultFilters)
                 if (className.length() == 0) {
                     className = rfClass.getName();
                 }
+                // we want to remember the order in which these were provided, and also keep the
+                // actual instances in case they have any additiona any state provided by the tool
+                // when they were created
                 toolDefaultReadFilterNamesInOrder.add(className);
                 toolDefaultReadFilters.put(className, f);
```

Comment:
What do you think about using a `LinkedHashMap` instead of a list and a `HashMap`, @cmnbroad?